cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsTableMigration.java

```diff
@@ -0,0 +1,206 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metrics.process;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.dataset2.lib.table.MetricsTable;
+import co.cask.cdap.data2.util.TableId;
+import co.cask.cdap.data2.util.hbase.HBaseDDLExecutorFactory;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.metrics.store.MetricDatasetFactory;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.spi.hbase.HBaseDDLExecutor;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.HBaseAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * perform table data migration from v2 metrics table to v3.
+ */
+public class MetricsTableMigration {
+  private static final Logger LOG = LoggerFactory.getLogger(MetricsTableMigration.class);
+
+  private final MetricsTable v2MetricsTable;
+  private final MetricsTable v3MetricsTable;
+  private final Configuration hConf;
+  private final CConfiguration cConf;
+
+  public MetricsTableMigration(MetricDatasetFactory metricDatasetFactory, int resolution,
+                               CConfiguration cConf, Configuration hConf) {
+    this(metricDatasetFactory.getV2MetricsTable(resolution),
+         metricDatasetFactory.getV3MetricsTable(resolution), cConf, hConf);
+  }
+
+  public MetricsTableMigration(MetricsTable v2MetricsTable, MetricsTable v3MetricsTable,
+                               CConfiguration cConf, Configuration hConf) {
+    this.v2MetricsTable = v2MetricsTable;
+    this.v3MetricsTable = v3MetricsTable;
+    this.cConf = cConf;
+    this.hConf = hConf;
+  }
+
+  /**
+   * returns true if old metrics data is still available for transfer
+   * @return
+   */
+  public boolean isOldMetricsDataAvailable() {
+    Scanner scanner = v2MetricsTable.scan(null, null, null);
+    Row row = scanner.next();
+    scanner.close();
+    return (row != null);
+  }
+
+  /**
+   * transfers data from v2 metrics table to v3 metrics table, limit on number of records is specified by
+   * maxRecordsToScan
+   * @param maxRecordsToScan - limit on number of records scanned
+   */
+  public void transferData(int maxRecordsToScan) {
+    LOG.trace("Running Metrics Migrator Thread");
```

Comment:
isn't this thread already running? You mean "running metrics migration for {} records", maxRecords?